packages/mermaid/src/diagrams/sankey/sankeyDB.ts

```diff
@@ -9,73 +7,63 @@ import {
   getDiagramTitle,
   clear as commonClear,
 } from '../../commonDb.js';
+import type { SankeyDiagramConfig } from '../../config.type.js';
+import type { SankeyDB, SankeyLink as ISankeyLink, SankeyFields } from './sankeyTypes.js';
+import DEFAULT_CONFIG from '../../defaultConfig.js';
+import type { RequiredDeep } from 'type-fest';
+
+export const DEFAULT_SANKEY_CONFIG: Required<SankeyDiagramConfig> = DEFAULT_CONFIG.sankey;
+
+export const DEFAULT_SANKEY_DB: RequiredDeep<SankeyFields> = {
+  links: [] as ISankeyLink[],
+  nodes: [] as string[],
+  config: DEFAULT_SANKEY_CONFIG,
+} as const;
 
 // Sankey diagram represented by nodes and links between those nodes
-let links: SankeyLink[] = [];
+let links: ISankeyLink[] = DEFAULT_SANKEY_DB.links;
 // Array of nodes guarantees their order
-let nodes: SankeyNode[] = [];
-// We also have to track nodes uniqueness (by ID)
-let nodesMap: Record<string, SankeyNode> = {};
+let nodes: string[] = DEFAULT_SANKEY_DB.nodes;
+const config: Required<SankeyDiagramConfig> = structuredClone(DEFAULT_SANKEY_CONFIG);
+
+const getConfig = (): Required<SankeyDiagramConfig> => structuredClone(config);
 
 const clear = (): void => {
   links = [];
   nodes = [];
-  nodesMap = {};
   commonClear();
 };
 
-class SankeyLink {
-  constructor(public source: SankeyNode, public target: SankeyNode, public value: number = 0) {}
-}
-
 /**
  * @param source - Node where the link starts
  * @param target - Node where the link ends
  * @param value - Describes the amount to be passed
  */
-const addLink = (source: SankeyNode, target: SankeyNode, value: number): void => {
-  links.push(new SankeyLink(source, target, value));
+const addLink = ({ source, target, value }: ISankeyLink): void => {
+  links.push({ source, target, value });
```

Comment:
If I am not mistaken we are destructuring attributes and combine them again. Do we need this?
```suggestion
const addLink = (link: ISankeyLink): void => {
  links.push(link);
```

Reply:
Because the `SankeyLink` from AST contains other unneeded attributes, if we destructure it would only pass needed ones.

https://github.com/mermaid-js/mermaid/blob/b6906a6474e01b5a6946749700150a4bada61657/packages/mermaid/src/diagrams/sankey/sankeyParser.ts#L12-L14